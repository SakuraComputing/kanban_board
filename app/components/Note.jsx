import React from 'react';

export default ({task}) => {
  return (
    <div>
      {task}
    </div>
  );
};

