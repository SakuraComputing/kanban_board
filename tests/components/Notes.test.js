import React from 'react';
import {shallow} from "enzyme/build/index";
import Note from '../../app/components/Note';

describe('Note Component', () => {
    let wrapper, notes, onValueClick, onEdit, onDelete;

    beforeEach(() => {
        notes = {
            id: 1,
            editing: true,
            task: 'wibble'
        };
        onValueClick = jest.fn();
        onEdit = jest.fn();
        onDelete = jest.fn();
        wrapper = shallow(<Note notes={notes} onValueClick={onValueClick} onEdit={onEdit} onDelete={onDelete}/>);
    });

    it('should exist', () => {
        expect(wrapper).toBeTruthy();
    });

    it('should render the form correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
