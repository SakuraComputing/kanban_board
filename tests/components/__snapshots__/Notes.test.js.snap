// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Component should render the form correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DragSource(DropTarget(Note)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "Consumer": [Circular],
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_changedBits": 0,
      "_changedBits2": 0,
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": Object {
        "dragDropManager": undefined,
      },
      "_currentValue2": Object {
        "dragDropManager": undefined,
      },
      "_defaultValue": Object {
        "dragDropManager": undefined,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_changedBits": 0,
        "_changedBits2": 0,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "dragDropManager": undefined,
        },
        "_currentValue2": Object {
          "dragDropManager": undefined,
        },
        "_defaultValue": Object {
          "dragDropManager": undefined,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
